### DISCUSSION TITLE
How to Use monaco-languageclient to Integrate Language Server for C++.

### DISCUSSION THREAD

**Discussion by navpreet032** [TypeFox/monaco-languageclient#548] [Q&A]
I was attempting to use Clangd as a language server for C++ in Monaco Editor. Based on my research, I found that I need to use vscode-ws-jsonrpc as a proxy between monaco-languageclient and Clangd. However, **I encountered difficulty in understanding how to set up this proxy and configure monaco-languageclient for C++**. If anyone has experience with this and knows how to do it, please provide guidance.

**Comment by kaisalmen** [comment]
Hi @navpreet032 you need to lauch the language server as external process as it is done with the pyright language server in our example:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/python/server/main.ts

And the client configuration is similar to the pyright client (But you need to adjust the configuration e.g. language id, extensions...):
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/python/server/main.ts

The problem is (and this applies to most more complex language servers implementation) that on the language client side you have to perform substantial implementation work to fully support the language server. The client implementation is specific and different for each language server. This is something `monaco-languagclient` can't do. It only supplies the glue to get you started.

Btw, we chose the python language server for our example, because it is fairly easy to integrate.

**Reply by navpreet032** [reply]
Is there a method to utilize the Clangd extension designed for VSCode  within Monaco Editor?.
